### DISCUSSION TITLE
Is it possible to use fragments within an SVG image?

### DISCUSSION THREAD

**Discussion by djelenc** [hakimel/reveal.js#3494] [Q&A] [answered]
I have an SVG image that I'd like to incrementally reveal its parts (maybe at some point even animate). 

I tried setting `class="fragment"` to its part but it did not work. (I include the image as a file, I do not embed) 

Can such thing be done?

**Comment by kjetilk** [comment, marked as answer]
I tried a plugin a long time ago, that I wasn't able to get working, but there [are others](https://github.com/marceloalcocer/reveal-svgfragments) that I haven't tried.

**Comment by jmckalex** [comment]
Yes, this is possible, but you have to embed the SVG graphic in the presentation. Since you have the SVG images as separate files, one way of doing this which is relatively painless is to use PHP, as follows:

```html
<p>
    Some text before the image.
    <?php echo file_get_contents("./path/to/file.svg"); ?>
    Some text after the image.
</p>
```

Alternatively, you could do this with JavaScript, as follows:

```html
<section>
<h3>title</h3>
<div id='foo'></div>
</section>
			
<script>
fetch('./path/to/file.svg')
  .then(response => response.text())
  .then(svgContent => {
	document.getElementById('foo').innerHTML = svgContent;
  });
</script>
```
